test/components/user_menu/signed_in_component_test.rb

```diff
@@ -60,6 +60,46 @@ class UserMenu::SignedInComponentTest < ViewComponent::TestCase
     assert_no_link_to admin_conversations_path
   end
 
+  test "links to new notifications view when no new notifications exist" do
+    user = users(:with_business)
+
+    render_inline UserMenu::SignedInComponent.new(user)
+    assert_link_to notifications_path
+  end
+
+  test "links to new notifications view when new notifications exist" do
+    user = users(:with_business)
+    developer = developers(:available)
+    Message.create!(developer: developer, business: user.business, sender: developer, body: "Hello!")
```

Comment:
We can use the `users(:with_business_conversation)` fixture instead of creating a new `Message` in the test. This also applies to the rest of this test file.

Reply:
I'm a little confused by this. I read the documentation on Rails unit testing but still have so many questions.

I see the users.yml file that defines the various fixtures. But I don't see anything creating any business conversations for the user "with_business_conversation", . How do I know that they have a new notification or a read notification? Does the testing suite create a copy of the database, so that any changes are not permanent? (ie if the test updates the db so that the new notification gets set as read, that won't affect future tests that depend on a new notification existing.)

Reply:
I'm new to fixtures, too, so we can learn together! I usually use FactoryBot, but am trying minitest/fixtures on this project because it is the default with Rails.

From my understanding, when the test suite starts each fixture is inserted into the database. Each test is also run in a transaction, so any db changes are reverted at the end of the test.

Figuring out which fixtures rely on others is a bit tricky, I'll admit. You kind of have to think of it inside-out. For example, to see the message for `users(:with_business_conversation)` I have to dig into `businesses(:with_conversation)` which is referenced from `conversations(:with_conversation)` which in turn is referenced from `messages(:from_business)`.

I just learned that [fixtures don't trigger callbacks](https://stackoverflow.com/a/1223192/384110). So on a test run there actually won't be any notifications getting created until you create a message manually. 

Does that make sense? I realize my initial advice was wrong!

Reply:
What are the action items for me on this one? 

Should I be trying to create fixtures for an unread notification and a read notification, and associating them with a message and a business or developer?

Reply:
Let's punt on this one for now. The tests pass and test the behavior - we can refactor later.